server/activityItem/queries.ts

```diff
@@ -0,0 +1,176 @@
+import * as types from 'types';
+
+import { Pub, ActivityItem, Collection, Member, Community, CollectionPub } from 'server/models';
+
+const createActivityItem = (ai: types.InsertableActivityItem) => ActivityItem.create(ai);
+
+const getDiffsForPayload = <
+	Entry extends Record<string, any>,
+	EntryKey extends keyof Entry,
+	Diffs = Partial<{ [Key in EntryKey]: types.Diff<Entry[Key]> }>
+>(
+	newEntry: Entry,
+	oldEntry: Entry,
+	keys: EntryKey[],
+): Diffs => {
+	return keys.reduce(
+		(memo: Diffs, key: EntryKey) =>
+			oldEntry[key] === newEntry[key]
+				? memo
+				: {
+						...memo,
+						[key]: {
+							from: oldEntry[key],
+							to: newEntry[key],
+						},
+				  },
+		{} as Diffs,
+	);
+};
+
+const getChangeFlagsForPayload = <
+	Entry extends Record<string, any>,
+	EntryKey extends keyof Entry,
+	Flags = Partial<{ [Key in EntryKey]: true }>
+>(
+	newEntry: Entry,
+	oldEntry: Entry,
+	keys: EntryKey[],
+): Flags => {
+	return keys.reduce(
+		(memo: Flags, key: EntryKey) =>
+			oldEntry[key] === newEntry[key] ? memo : { ...memo, [key]: true },
+		{} as Flags,
+	);
+};
+
+export const createPubActivityItem = async (
+	kind: 'pub-created' | 'pub-updated' | 'pub-removed' | 'pub-released',
```

Comment:
`pub-released` probably justifies a separate function here. It's the only one of these where the `releaseId` argument is required, which is possible(?) but grisly to express in TS. I'd suggest removing it from this list along with the `releaseId` parameter and adding this functionality to the second batch.